powerdns.go

```diff
@@ -134,9 +135,8 @@ func (p *Client) newRequest(method string, path string, query *url.Values, body
 
 	if body != nil {
 		req.Header.Set("Content-Type", "application/json")
-		req.Header.Set("Accept", "application/json")
 	}
-
+	req.Header.Set("Accept", "application/json")
```

Comment:
`zones/export` usually returns a plain text value. After setting this header globally, PowerDNS 4.2 will return a JSON object instead, hence break the `Export` function.

You can reproduce this if you run the unit tests without mocks. Start the Docker environment with `docker-compose -f docker-compose-v4.2.yml up`, wait a moment until PowerDNS finished initialising, and then run `make test-without-mocks`. Perhaps we could use this opportunity to also improve the mocks for `TestExport`.

Reply:
Yes, have not noticed that export has to return string, you are totally right. 
TestExport mock extended. Test will fail if to move *req.Header.Set("Accept", "application/json")* out of "if" statement.